findDate.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Searches for dates in an array of strings. The dates can be in form
+ * dd.mm.yyyy, dd.mm.yy or yyyy-mm-dd
+ *
+ * @param {string[]} fragments array of string fragments to be scanned for dates
+ * @returns the date in the input array as js Date object or null, if no strings contain a date. If multiple dates are found, returns the date closest to today.
+ */
+function findDate(fragments) {
+  let ret = null;
+  let matches = [];
+  for (let i = 0; i < fragments.length; i += 1) {
+    const m1 = fragments[i].match(
+      /[0-9][0-9]\.[0-9][0-9]\.[0-9][0-9][0-9][0-9]/g
+    );
+    const m2 = fragments[i].match(/[0-9][0-9]\.[0-9][0-9]\.[0-9][0-9]/g);
+    const m3 = fragments[i].match(
+      /[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]/g
+    );
+    if (m1 && m1.length > 0) {
+      matches = matches.concat(m1);
+    }
+    if (m2 && m2.length > 0) {
+      matches = matches.concat(m2);
+    }
+    if (m3 && m3.length > 0) {
+      matches = matches.concat(m3);
+    }
+  }
+
```

Comment:
Here, please add a function to check if all the input dates are valid. Any invalid date should not be passed further.
Pseudo code:
let newMatches = isValid(matches)

function isValid(matches) {
  This function should return only valid dates
}

Use newMatches for further validation